src/components/NavBar/NavBar.js

```diff
@@ -27,23 +30,62 @@ export default function ResponsiveAppBar() {
 	const handleOpenNavMenu = (event) => {
 		setAnchorElNav(event.currentTarget);
 	};
-	const handleOpenUserMenu = (event) => {
-		setAnchorElUser(event.currentTarget);
-	};
-
 	const handleCloseNavMenu = () => {
 		setAnchorElNav(null);
 	};
 
+	const handleOpenUserMenu = (event) => {
+		setAnchorElUser(event.currentTarget);
+	};
 	const handleCloseUserMenu = () => {
 		setAnchorElUser(null);
 	};
 
+	const router = useRouter();
+	const auth = getAuth();
+	const login = () => {
```

Comment:
i don't think it makes sense for the NavBar to have so much firebase code - can this be a function that's exported from the `firebase` section and NavBar uses?

Reply:
That's a good idea. Moved code to firebase folder.

Reply:
Actually I couldn't find a way to do this perfectly so far, because there was some problem with the useRouter hook. It can only be used in a component and I couldn't find a way to use it in an external function. I will try later after finishing other prioritized functions.